src/Microsoft.ML.Tokenizers/Model/SentencePieceBpeModel.cs

```diff
@@ -41,6 +41,52 @@ internal SentencePieceBpeModel(ModelProto modelProto, bool addBos, bool addEos,
             OneByteUtf8EncodingMaxId = ByteCodeToIdOffset + 0x7F; // 0x7F is the maximum value of the one byte UTF-8 character.
         }
 
+        internal SentencePieceBpeModel(SentencePieceOptions options) : base(options)
+        {
+            if (options.PrecompiledNormalizationData is not null)
+            {
+                throw new NotSupportedException("Normalization data is not supported for SentencePieceBpeModel.");
+            }
+
+            Debug.Assert(options.Vocabulary is not null);
+
+            int id = 0;
+            foreach (var item in options.Vocabulary!)
+            {
```

Comment:
Can we check for the sort requirement here while we initialize it and throw if the data violates that requirement?

Reply:
We cannot. We don't have anything to compare against. The sort order  is the order the entries listed in the data file. it is not sorting tokens or numbers.

Reply:
Oh, I see.  That wasn't clear to me from the docs.  So the index in the list is the sort.